Add list semantics to `AvatarGroup` (#1437)
iTwin/stratakit: apps/website/src/content/docs/components/mui/AvatarGroup.md

- DOM order now matches the visual display order.
- The visual stacking of the `Avatar`s is inverted.

> - Added list semantics to `AvatarGroup` and nested `Avatar`s. This required adding a wrapping element on each `Avatar`.

Reviewer: ```suggestion
- Added list semantics to `AvatarGroup`. Each child `Avatar` is wrapped in a list item element.
```
